### PR TITLE
Upload reanalyze template merge

### PR DESCRIPTION
**Pull request by jstucke**
- :warning:  is branched from `add_route_decorator` ->  #488 should be merged first
- merged upload and re-analyze templates (>90% identical)
- moved `overwrite_default_plugins()` helper function (only used once)

### CODE REVIEW

File: src/test/unit/web_interface/test_app_re_analyze.py [hunk 1]
```diff
@@ -32,12 +32,13 @@ def test_app_re_analyze_post_valid(self):
         rv = self.test_client.post('/update-analysis/{}'.format(TEST_FW.uid), data=form_data)
         assert b'Upload Successful' in rv.data
         assert make_bytes(TEST_FW.uid) in rv.data
-        self.assertEqual(self.mocked_interface.tasks[0].uid, TEST_FW.uid, "fw not added to intercom")
-        self.assertIn("new_system", self.mocked_interface.tasks[0].scheduled_analysis, "new analysis system not scheduled")
+        assert self.mocked_interface.tasks[0].uid == TEST_FW.uid, "fw not added to intercom"
```
Comment (dorpvom) [inline src/test/unit/web_interface/test_app_re_analyze.py:35]: Mix of `'` and `"` quotation

File: src/web_interface/templates/upload/upload.html [hunk 2]
```diff
@@ -30,229 +37,212 @@ <h3 class="mb-3">Upload Firmware</h3>
...
                                 for (var key in device_classes) {
                                     if (device_classes.hasOwnProperty(key)) {
-                                        $('#device_name_dropdown').append('<option>' + device_classes[key] + '</option>');
+                                        $('#device_name').append('<option>' + device_classes[key] + '</option>');
                                     }
                                 }
                             }
                         }
-                        $('#device_name_dropdown').append('<option>' + 'new entry' + '</option>');
+                        $('#device_name').append('<option>' + 'new entry' + '</option>');
```
Comment (dorpvom) [inline src/web_interface/templates/upload/upload.html:65]: At about this point in the file I now decided to not code review this file further.
I'll make a functional review and leave it at that.

**Review by dorpvom — CHANGES_REQUESTED** [review]

File: src/web_interface/templates/upload/upload.html [hunk 3]
```diff
@@ -1,8 +1,11 @@
-{% extends "base.html" %}
+{%- extends "base.html" -%}
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/web_interface/templates/upload/upload.html:1]: The purpose of these is to remove the whitespace when rendering the template.
E.g.
```html
<tag1>
    {% if foo %}
    {% endif %}
        <tag2>
```
would render to
```html
<tag1>


        <tag2>
```
and

```html
<tag1>
    {%- if foo -%}
    {%- endif %}
        <tag2>
```
would render to
```html
<tag1>
        <tag2>
```
Long story short: the source of our templates looks less messy.

File: src/web_interface/templates/upload/upload.html [hunk 4]
```diff
@@ -30,229 +37,212 @@ <h3 class="mb-3">Upload Firmware</h3>
...
+                        <input class="form-control-file" id="file" type="file" name=file required>
+                    {% endcall %}
+                {%- else %}
+                    <input type="hidden" name="file_name" id="file_name" value="{{ firmware.file_name }}">
+                {%- endif %}
 
                 <script>
-                    var device_class_dropdown = document.getElementById('device_class_dropdown');
-                    var vendor_dropdown = document.getElementById('vendor_dropdown');
+                    var device_names = eval( {{ device_names | safe }} );
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/web_interface/templates/upload/upload.html:50]: If I remember correctly, this was done to pass json from jinja to javascript but it turns out it isn't even necessary -> removed

File: src/test/unit/web_interface/test_app_re_analyze.py (hunk 1 above)
Comment (jstucke) [reply src/test/unit/web_interface/test_app_re_analyze.py:35]: fixed

**Review by dorpvom — APPROVED** [review]

**Pull request merged**